### PR TITLE
Add an `undertow-benchmarks` module with basic JMH benchmarks

### PR DESCRIPTION
**Pull request by carterkozak**
Currently this only provides four simple benchmarks for blocking I/O
functionality, but may be expanded to benchmark additional features.

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
Can one of the admins verify this patch?

File: benchmarks/src/main/resources/logging.properties [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+formatter.PATTERN.properties=pattern
+formatter.PATTERN.pattern=%d{HH:mm:ss,SSS} %-5p (%t) [%c] <%F:%L> %m%n
+
+logger.org.xnio.listener.level=DEBUG
+
+logger.org.xnio.ssl.level=DEBUG
+
+logger.org.apache.level=WARN
+logger.org.apache.useParentHandlers=false
+logger.io.undertow.util.TestHttpClient.level=WARN
```
Comment (carterkozak) [inline benchmarks/src/main/resources/logging.properties]: This logger configuration and the certs are borrowed from undertow-core tests

File: benchmarks/pom.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2012 Red Hat, Inc., and individual contributors
```
Comment (rhusar) [inline benchmarks/pom.xml]: 2019 :-)

**Comment by rhusar** [comment]
this is ok to test

File: benchmarks/pom.xml (hunk 2 above)
Comment (carterkozak) [reply benchmarks/pom.xml]: Oops, fixed

**Comment by undertow-pull-request** [comment]
**Linux** [Build 2995](https://ci.wildfly.org/viewLog.html?buildId=147438&buildTypeId=UT_MASTER_LINUX) outcome was **FAILURE** using a merge of c5a79c42747e12eaa24ecbd541c62c4e93799159
Summary: Tests passed: 3463, ignored: 632; exit code 1 (Step: Maven) (new) Build time: 00:39:06

**Comment by undertow-pull-request** [comment]
**Windows** [Build 2747](https://ci.wildfly.org/viewLog.html?buildId=147439&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of c5a79c42747e12eaa24ecbd541c62c4e93799159
Summary: Tests failed: 2 (1 new), passed: 3383, ignored: 710 Build time: 00:45:01
### Failed tests
```
io.undertow.websockets.jsr.test.annotated.AnnotatedEndpointTest.testRequestUri: java.lang.IllegalStateException: ChannelPipeline does not contain a HttpRequestEncoder or HttpClientCodec
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker$1.operationComplete(WebSocketClientHandshaker.java:182)
	at io.netty.handler.codec.http.websocketx.WebSocketClientHandshaker$1.operationComplete(WebSocketClientHandshaker.java:172)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:481)
	at io.netty.util.concurrent.DefaultPromise.access$000(DefaultPromise.java:34)
	at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:431)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:748)
------- Stderr: -------
16:34:46,960 DEBUG (XNIO-1 task-17) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /ws/request, authentication required: false
16:34:46,960 DEBUG (XNIO-1 task-17) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@45d65eb0 for /ws/request
16:34:46,960 DEBUG (XNIO-1 task-17) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /ws/request


io.undertow.server.handlers.ReceiverTestCase.testAsyncReceiveWholeBytesFailed[proxy][http2]: java.lang.AssertionError
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceiveWholeBytesFailed(ReceiverTestCase.java:194)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:528)
	at io.undertow.testutils.DefaultServer.runChild(DefaultServer.java:104)
	at io.undertow.testutils.DefaultServer.run(DefaultServer.java:301)
------- Stderr: -------
16:03:57,734 DEBUG (XNIO-1 I/O-1) [io.undertow.server.handlers.proxy] <ProxyHandler.java:549> Sending request ClientRequest{path='/fullbytes', method=POST, protocol=HTTP/1.1} to target localhost/127.0.0.1:8888 for exchange HttpServerExchange{ POST /fullbytes}
16:03:57,735 DEBUG (XNIO-1 I/O-1) [io.undertow.server.handlers.proxy] <ProxyHandler.java:566> Sent request ClientRequest{path='/fullbytes', method=POST, protocol=HTTP/1.1} to target 127.0.0.1 for exchange HttpServerExchange{ POST /fullbytes}
16:03:57,793 ERROR (XNIO-1 I/O-1) [io.undertow.proxy] <ProxyHandler.java:662> UT005028: Proxy request to /fullbytes failed: java.nio.channels.ClosedChannelException
	at io.undertow.client.http2.Http2ClientConnection$1.handleEvent(Http2ClientConnection.java:107)
	at io.undertow.client.http2.Http2ClientConnection$1.handleEvent(Http2ClientConnection.java:99)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameCloseListener.handleEvent(AbstractFramedChannel.java:1061)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameCloseListener.handleEvent(AbstractFramedChannel.java:975)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.terminated(SslConduit.java:1214)
	at org.xnio.nio.NioSocketConduit.readTerminated(NioSocketConduit.java:360)
	at org.xnio.nio.NioSocketStreamConnection.notifyReadClosed(NioSocketStreamConnection.java:148)
	at org.xnio.Connection.close(Connection.java:139)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.protocols.ssl.SslConduit.closed(SslConduit.java:1046)
	at io.undertow.protocols.ssl.SslConduit.close(SslConduit.java:1117)
	at io.undertow.protocols.ssl.UndertowSslConnection.closeAction(UndertowSslConnection.java:149)
	at org.xnio.Connection.close(Connection.java:132)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.server.protocol.framed.AbstractFramedChannel.close(AbstractFramedChannel.java:816)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.protocols.http2.Http2Channel.sendGoAway(Http2Channel.java:832)
	at io.undertow.protocols.http2.Http2Channel.sendGoAway(Http2Channel.java:809)
	at io.undertow.client.http2.Http2ClientConnection.close(Http2ClientConnection.java:303)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.server.handlers.proxy.ProxyHandler$IoExceptionHandler.handleException(ProxyHandler.java:790)
	at org.xnio.ChannelListeners.invokeChannelExceptionHandler(ChannelListeners.java:126)
	at io.undertow.util.Transfer$TransferListener.handleEvent(Transfer.java:218)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedStreamSinkChannel$1.run(AbstractFramedStreamSinkChannel.java:214)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:935)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:929)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.readReady(SslConduit.java:1175)
	at io.undertow.protocols.ssl.SslConduit$1.run(SslConduit.java:168)
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:582)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:466)

16:03:57,793 DEBUG (XNIO-1 I/O-1) [io.undertow.request.error-response] <DelegatingBasicLogger.java:424> Setting error code 503 for exchange HttpServerExchange{ POST /fullbytes}: java.lang.RuntimeException
	at io.undertow.server.HttpServerExchange.setStatusCode(HttpServerExchange.java:1410)
	at io.undertow.server.handlers.proxy.ProxyHandler.handleFailure(ProxyHandler.java:668)
	at io.undertow.server.handlers.proxy.ProxyHandler$ResponseCallback.failed(ProxyHandler.java:727)
	at io.undertow.client.http2.Http2ClientExchange.failed(Http2ClientExchange.java:125)
	at io.undertow.client.http2.Http2ClientConnection$1.handleEvent(Http2ClientConnection.java:107)


```

**Comment by fl4via** [comment]
Retest this please

**Comment by fl4via** [comment]
Retest this please

**Comment by fl4via** [comment]
Retest this please

**Comment by carterkozak** [comment]
I think the build nodes look sad, I don't think these errors are related to my change.

**Pull request merged**